### PR TITLE
Login

### PR DESCRIPTION
**Pull request by orcunsagirsoy**
Screenshot from 2020-11-12 01-47-23

Login page:
-Provides logging in of a user to the website, the distinction between a non-user and a user will be defined later
-Contains Google and Facebook logins, Facebook login button style has problems, editing it is quite not possible any help would be appreciated
-User can click and reset their password, also if already has an account user may proceed to the sign-up page
-Login component has firebase authentication

Screenshot from 2020-11-12 01-48-03

Sign-up page:
-Sends email and password data to Firebase
-Name and surname data will be sent to Firebase after the user profile object has been created
-If the user has an account, they can proceed to login page

Screenshot from 2020-11-12 01-48-21

Password reset page:
-User can proceed to the Login and Signup pages if wanted
-Sends an email to the user about resetting the password

Other:
-PriveRoute component provides access to updating email and password if the user logged in
-Google logout component is not in use, it is for further usage
-AuthContext component provides necessary functions for authentication
-Routing has been established in App.jsx
-utils folder has a token refreshing function for every one hour
-translation object has been updated
-The image used in three pages is located in the assets folder
-Test files are missing and will be added when they finished

### CODE REVIEW

**Comment by Luey-A** [comment]
i think it would look better if you can move the pic at the bottom to the center of the page. it is too close to the left.

**Comment by Ammar-64** [comment]
@orcunsagirsoy can you merge dev with your branch, there are too many changes in the files, also make sure the tests are passing, so I can review it. And connect the login and sign up pages to the buttons in the navbar

**Comment by orcunsagirsoy** [comment]
PTAL

File: src/App.jsx [hunk 1]
```diff
@@ -1,24 +1,36 @@
...
-      <Route path="/" component={SiteNavbar} />
-      <Route exact path="/" component={Home} />
-      <Route path="/about" component={About} />
-      <Route path="/" component={Footer} />
-      <Route exact path="/donate" component={NewPostForm} />
-      {/* <Route path="/posts" component={Posts} />
-      <Route path="/donate" component={Donate} />
-      <Route path="/contact" component={Contactus} /> */}
-    </Router>
+    <>
```
Comment (Ammar-64) [inline src/App.jsx:15]: You don’t need to wrap them with a div and <>, <AuthProvider> Is enough

File: src/components/LoginPage/FacebookLogin/index.jsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+class Login extends React.Component {
+  state = { user: false };
+  facebookResponse = (response) => {
+    console.log(response);
+    this.setState({ ...this.state, user: response });
+  };
+
+  render() {
+    return (
+      <div style={{ margin: "auto", textAlign: "center", width: "13em" }}>
```
Comment (Ammar-64) [inline src/components/LoginPage/FacebookLogin/index.jsx]: Please add the style in a separate css file

File: src/components/LoginPage/ForgotPassword/index.css [hunk 3]
```diff
@@ -0,0 +1,17 @@
+.ressPassBtn {
+  background-color: #c62a88;
+  border: none;
+}
+
+.purple {
```
Comment (Ammar-64) [inline src/components/LoginPage/ForgotPassword/index.css]: Can you give more unique names for theses classNames, so the reader would understand what are they referring to

File: src/components/LoginPage/ForgotPassword/index.jsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    } catch {
+      setError("Failed to reset password");
+    }
+    setLoading(false);
+  }
+
+  return (
+    <>
+      <Container
+        className="d-flex flex-column justify-content-center"
```
Comment (Ammar-64) [inline src/components/LoginPage/ForgotPassword/index.jsx:36]: Also style in a separat css

File: src/components/LoginPage/GoogleLogin/index.css [hunk 5]
```diff
@@ -0,0 +1,6 @@
+.buttonG {
```
Comment (Ammar-64) [inline src/components/LoginPage/GoogleLogin/index.css]: Give a more meaningful name to this className

File: src/components/LoginPage/GoogleLogout/index.jsx [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  };
+
+  const { signOut } = useGoogleLogout({
+    clientId,
+    onLogoutSuccess,
+    onFailure,
+  });
+
+  return (
+    <button onClick={signOut} className="button">
```
Comment (Ammar-64) [inline src/components/LoginPage/GoogleLogout/index.jsx]: Give more meaningful className, button might e used by a library or a different team mate and it’ll make conflicts

File: src/components/LoginPage/GoogleLogout/index.jsx [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+      <img src="./icons/google.svg" alt="google logout" className="icon"></img>
```
Comment (Ammar-64) [inline src/components/LoginPage/GoogleLogout/index.jsx]: Same for icon className

File: src/components/LoginPage/Login/index.css [hunk 8]
```diff
@@ -0,0 +1,13 @@
+.logInBtn {
+  background-color: #c62a88;
+  border: none;
+}
+
+.purple {
```
Comment (Ammar-64) [inline src/components/LoginPage/Login/index.css]: classNames here to

File: src/components/LoginPage/Login/index.jsx [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+    } catch {
+      setError("Failed to log in");
+    }
+
+    setLoading(false);
+  }
+
+  return (
+    <>
+      <Card className="shadow" style={{ marginTop: "10em" }}>
```
Comment (Ammar-64) [inline src/components/LoginPage/Login/index.jsx]: classNames

File: src/components/LoginPage/Logout/index.css [hunk 10]
```diff
@@ -0,0 +1,13 @@
+.logOutBtn {
+  background-color: #c62a88;
+  border: none;
+}
+
+.purple {
```
Comment (Ammar-64) [inline src/components/LoginPage/Logout/index.css]: Same className

**Review by Ammar-64 — CHANGES_REQUESTED** [review]
Solid work but can you please take a look at the comments about classNames, if you’re using the same classes for all of them you can import it to the container and add them in css there

File: src/components/LoginPage/Logout/index.css (hunk 10 above)
Comment (orcunsagirsoy) [reply src/components/LoginPage/Logout/index.css]: PTAL

File: src/components/LoginPage/Login/index.jsx (hunk 9 above)
Comment (orcunsagirsoy) [reply src/components/LoginPage/Login/index.jsx]: PTAL

File: src/components/LoginPage/GoogleLogout/index.jsx (hunk 7 above)
Comment (orcunsagirsoy) [reply src/components/LoginPage/GoogleLogout/index.jsx]: PTAL

File: src/components/LoginPage/GoogleLogout/index.jsx (hunk 6 above)
Comment (orcunsagirsoy) [reply src/components/LoginPage/GoogleLogout/index.jsx]: PTAL

File: src/components/LoginPage/ForgotPassword/index.jsx (hunk 4 above)
Comment (orcunsagirsoy) [reply src/components/LoginPage/ForgotPassword/index.jsx:36]: PTAL

File: src/components/LoginPage/ForgotPassword/index.css (hunk 3 above)
Comment (orcunsagirsoy) [reply src/components/LoginPage/ForgotPassword/index.css]: PTAL

File: src/components/LoginPage/FacebookLogin/index.jsx (hunk 2 above)
Comment (orcunsagirsoy) [reply src/components/LoginPage/FacebookLogin/index.jsx]: PTAL

**Review by betulerkoc — APPROVED** [review]

File: src/components/LoginPage/FacebookLogin/index.jsx [hunk 11]
```diff
@@ -0,0 +1,53 @@
+import React from "react";
+import "./index.css";
+import FacebookLoginWithButton from "react-facebook-login";
+
+const componentClicked = () => {
+  console.log("Clicked!");
```
Comment (Ammar-64) [inline src/components/LoginPage/FacebookLogin/index.jsx]: Console.log shouldn’t be in production

File: src/components/LoginPage/FacebookLogin/index.jsx [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+    <img
+      src={user.picture.data.url}
+      height={user.picture.height}
+      width={user.picture.width}
+      alt="avatar"
+    />
+  </>
+);
+
+class Login extends React.Component {
```
Comment (Ammar-64) [inline src/components/LoginPage/FacebookLogin/index.jsx]: Why is this class component it should be functional component

File: src/components/LoginPage/GoogleLogin/index.jsx [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+import "./index.css";
+import { useTranslation } from "react-i18next";
+
+const clientId =
+  "743068888490-2chsv38s7paclkbc1qjd2ktd3jjf2aqf.apps.googleusercontent.com";
+
+function Login() {
+  const { t } = useTranslation();
+  const onSuccess = (res) => {
+    console.log("[Login Success] currentUser:", res.profilObj);
```
Comment (Ammar-64) [inline src/components/LoginPage/GoogleLogin/index.jsx]: No console.log should be in production

File: src/components/LoginPage/GoogleLogout/index.jsx [hunk 14]
```diff
@@ -0,0 +1,36 @@
+import React from "react";
+import { useGoogleLogout } from "react-google-login";
+
+const clientId =
+  "743068888490-2chsv38s7paclkbc1qjd2ktd3jjf2aqf.apps.googleusercontent.com";
+
+function Logout() {
+  const onLogoutSuccess = (res) => {
+    console.log("Logged out Success");
```
Comment (Ammar-64) [inline src/components/LoginPage/GoogleLogout/index.jsx]: console.log

File: src/components/LoginPage/GoogleLogout/index.jsx [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+    alert("Logged out Successfully");
+  };
+
+  const onFailure = () => {
+    console.log("Handle failure cases");
```
Comment (Ammar-64) [inline src/components/LoginPage/GoogleLogout/index.jsx]: Same here

File: src/components/LoginPage/Login/index.jsx (hunk 9 above)
Comment (Ammar-64) [reply src/components/LoginPage/Login/index.jsx]: What does shadow mean, could it be more meaningful?

File: src/components/LoginPage/Logout/index.jsx [hunk 16]
```diff
@@ -0,0 +1,43 @@
...
+        className="btn btn-primary w-100 mt-3"
+      >
+        Update Profile
+      </Link>
+      <div className="w-100 text-center mt-2">
+        <Button
+          variant="link logOutBtn purpleElemLogout"
+          onClick={handleLogout}
+        >
+          Log Out
```
Comment (Ammar-64) [inline src/components/LoginPage/Logout/index.jsx]: All texts should be coming from i18next

File: src/containers/Login/index.css [hunk 17]
```diff
@@ -0,0 +1,10 @@
+.googlefacebookLogin {
```
Comment (Ammar-64) [inline src/containers/Login/index.css]: Can this be camelCaseed i.e. googleFacebookLogIn

File: src/components/LoginPage/UpdateLoginInfo/index.jsx [hunk 18]
```diff
@@ -0,0 +1,110 @@
...
+                  ref={passwordConfirmRef}
+                  placeholder="Leave blank to keep the same"
+                />
+              </Form.Group>
+              <Button
+                disabled={loading}
+                className="w-100 updateBtn"
+                type="submit"
+              >
+                Update
```
Comment (Ammar-64) [inline src/components/LoginPage/UpdateLoginInfo/index.jsx]: Texts should be coming from i18next

File: src/components/LoginPage/Signup/index.jsx [hunk 19]
```diff
@@ -0,0 +1,121 @@
...
+    if (passwordRef.current.value !== passwordConfirmRef.current.value) {
+      return setError("Passwords do not match");
+    }
+    try {
+      setError("");
+      setLoading(true);
+      await signup(emailRef.current.value, passwordRef.current.value);
+      history.push("/");
+    } catch {
+      setError("Failed to create an account");
```
Comment (Ammar-64) [inline src/components/LoginPage/Signup/index.jsx]: Texts should be coming from i18next

File: src/components/LoginPage/Login/index.jsx [hunk 20]
```diff
@@ -0,0 +1,78 @@
...
+              </Form.Label>
+              <Form.Control
+                className="purpleElemLogin"
+                type="password"
+                ref={passwordRef}
+                required
+              />
+            </Form.Group>
+            <Button disabled={loading} className="w-100 logInBtn" type="submit">
+              Log In
```
Comment (Ammar-64) [inline src/components/LoginPage/Login/index.jsx]: Texts should be coming from i18next

File: src/components/LoginPage/GoogleLogout/index.jsx [hunk 21]
```diff
@@ -0,0 +1,36 @@
...
+
+  return (
+    <button onClick={signOut} className="googleLogOutBtn">
+      <img
+        src="./icons/google.svg"
+        alt="google logout"
+        className="googleIcon"
+      ></img>
+
+      <span className="buttonText">Sign out</span>
```
Comment (Ammar-64) [inline src/components/LoginPage/GoogleLogout/index.jsx]: Texts should be coming from i18next

**Review by Ammar-64 — CHANGES_REQUESTED** [review]
Good work @orcunsagirsoy some extra things need to be changed and we’re good to go

File: src/components/LoginPage/Login/index.jsx (hunk 9 above)
Comment (orcunsagirsoy) [reply src/components/LoginPage/Login/index.jsx]: that is bootstrap class, should I write it manually?

**Comment by orcunsagirsoy** [comment]
PTAL, I removed the facebook login, for now, will work on it later. I transformed all text to i18 files and fixed class names issues

File: src/components/LoginPage/GoogleLogin/index.jsx [hunk 22]
```diff
@@ -0,0 +1,49 @@
...
+    accessType: "offline",
+  });
+
+  return (
+    <button onClick={signIn} className="buttonGoogle">
+      <div>
+        <img
+          src={googleIcon}
+          alt="google login"
+          className="icon"
```
Comment (Ammar-64) [inline src/components/LoginPage/GoogleLogin/index.jsx]: Is this className being used? I couldn’t find where, if yes can you change the name of it please

File: src/components/LoginPage/Login/index.jsx (hunk 9 above)
Comment (Ammar-64) [reply src/components/LoginPage/Login/index.jsx]: No worries just add a comment to indicate that it’s a bootstrap class

**Review by Ammar-64 — COMMENTED** [review]
Good job, small extra stuff and we’re good to go

**Comment by Ammar-64** [comment]
@orcunsagirsoy after I log in how do I know it works, I logged but couldn’t see if I’m in or not, and didn’t see a log out afterwards

**Comment by Ammar-64** [comment]
@orcunsagirsoy instead of adding a new button to logout and the email, can you remove the signup and log in buttons and replace them with logout with the name of the person instead of the email, does it make sense, if you want I can help you with this

File: src/components/LoginPage/ForgotPassword/index.jsx [hunk 23]
```diff
@@ -0,0 +1,86 @@
...
+
+  async function handleSubmit(e) {
+    e.preventDefault();
+
+    try {
+      setMessage("");
+      setError("");
+      setLoading(true);
+      await resetPassword(emailRef.current.value);
+      setMessage("Check your inbox!");
```
Comment (Ammar-64) [inline src/components/LoginPage/ForgotPassword/index.jsx]: Message and error are texts and should be coming from i18next

File: src/components/Navbar/index.jsx [hunk 24]
```diff
@@ -46,8 +67,13 @@ export default function SiteNavbar() {
         </Form>
 
         <Button className="navButtons">{t("navbarsection.Search")}</Button>
-        <Button className="navButtons">{t("navbarsection.LogIn")}</Button>
-        <Button className="navButtons signUpButton">
```
Comment (Ammar-64) [inline src/components/Navbar/index.jsx:50]: What about doing this here ` {currentUser ? (
           <div className="loggedInUserInfo">
             Logged in as {currentUser.name}
           </div>  <Logout />
         ):   <Button onClick={handleClickLogin} className="navButtons">
           {t("navbarsection.LogIn")}
         </Button>
         <Button
           onClick={handleClickSignup}
           className="navButtons signUpLogoutButton"
         >
           {t("navbarsection.SignUp")}
         </Button>}
        } `

**Review by Ammar-64 — COMMENTED** [review]
Can you have a look at the comment and let me know what you think

File: src/components/Navbar/index.jsx (hunk 24 above)
Comment (orcunsagirsoy) [reply src/components/Navbar/index.jsx:50]: Yeah I can handle the buttons, but showing user name I need to think about it cuz I dont have an idea right now

**Comment by orcunsagirsoy** [comment]
x-special/nautilus-clipboard
copy
file:///home/orcun/Desktop/Screenshot%20from%202020-11-20%2019-51-57.png

-New version of login pages, I think login pages needed to be more colorized.
-Logged in and logged out navbar contains different buttons now
-Login and google login buttons are same size.

**Comment by orcunsagirsoy** [comment]
Screenshot from 2020-11-20 19-51-57

-New version of login pages, I think login pages needed to be more colorized.
-Logged in and logged out navbar contains different buttons now
-Login and google login buttons are same size.

Screenshot from 2020-11-20 20-05-20

File: src/components/LoginPage/ForgotPassword/index.jsx (hunk 23 above)
Comment (orcunsagirsoy) [reply src/components/LoginPage/ForgotPassword/index.jsx]: PTAL

File: src/components/Navbar/index.jsx (hunk 24 above)
Comment (orcunsagirsoy) [reply src/components/Navbar/index.jsx:50]: PTAL

File: src/components/LoginPage/ForgotPassword/index.css [hunk 25]
```diff
@@ -0,0 +1,24 @@
...
+.greenElemForgotPass {
+  color: #03c4a1;
+}
+
+.friendsBoxImg {
+  margin-top: 1em;
+}
+
+.neumorphism {
+  border-radius: 10px;
```
Comment (Ammar-64) [inline src/components/LoginPage/ForgotPassword/index.css]: px should be replaced with em

File: src/components/LoginPage/GoogleLogin/index.jsx [hunk 26]
```diff
@@ -0,0 +1,49 @@
...
+  });
+
+  return (
+    <button onClick={signIn} className="buttonGoogle">
+      <div>
+        <img
+          src={googleIcon}
+          alt="google login"
+          className="googleIcon"
+          width="40em"
```
Comment (Ammar-64) [inline src/components/LoginPage/GoogleLogin/index.jsx:39]: Is this responsive ?

File: src/components/LoginPage/Login/index.css [hunk 27]
```diff
@@ -0,0 +1,33 @@
...
+  color: #590995;
+  border-color: #590995;
+}
+
+.greenElemLogin {
+  color: #03c4a1;
+}
+
+.neumorphism {
+  border-radius: 10px;
```
Comment (Ammar-64) [inline src/components/LoginPage/Login/index.css]: Px should be replaced with em

File: src/components/LoginPage/Signup/index.css [hunk 28]
```diff
@@ -0,0 +1,24 @@
...
+.greenElemSignUp {
+  color: #03c4a1;
+}
+
+.friendsBoxImg {
+  margin-top: 1em;
+}
+
+.neumorphism {
+  border-radius: 10px;
```
Comment (Ammar-64) [inline src/components/LoginPage/Signup/index.css]: Px should be replaced with em

**Review by Ammar-64 — COMMENTED** [review]
LGTM, some really small changes and everything is good to go

File: src/components/LoginPage/GoogleLogin/index.jsx (hunk 26 above)
Comment (orcunsagirsoy) [reply src/components/LoginPage/GoogleLogin/index.jsx:39]: does it need to get smaller when screen is small? I did not understand how a button icon could be responsive?

Comment (orcunsagirsoy) [reply src/components/LoginPage/GoogleLogin/index.jsx:39]: its width is 3em, so I guess it is responsive

**Review by Ammar-64 — APPROVED** [review]
LGTM

**Pull request merged**